flake.nix

```diff
@@ -21,29 +27,53 @@
           overlays = [
             haskellNix.overlay
             (final: prev: {
-              hapistrano = final.haskell-nix.cabalProject' {
+              hapistrano-ghc8107 = final.haskell-nix.cabalProject' {
                 src = final.haskell-nix.haskellLib.cleanGit {
                   name = "hapistrano";
                   src = ./.;
                 };
                 compiler-nix-name = "ghc8107";
               };
+              hapistrano-ghc902 = final.haskell-nix.cabalProject' {
+                src = final.haskell-nix.haskellLib.cleanGit {
+                  name = "hapistrano";
+                  src = ./.;
+                };
+                compiler-nix-name = "ghc902";
+              };
             })
           ];
         };
-        flake = pkgs.hapistrano.flake { };
+        flake-ghc8107 = pkgs.hapistrano-ghc8107.flake { };
+        flake-ghc902 = pkgs.hapistrano-ghc902.flake { };
       in
-      flake // rec {
+      flake-ghc8107 // flake-ghc902 // rec {
```

Comment:
@oscar-izval These two effectively overwrite each other; I believe it would be better to expose a minimal flake output based on some specific use cases, such as "apps.test-*" to run the tests and "packages.test-*" for reference on run, and perhaps "devShells" to allow maintainers to spawn a new shell to work on this project. Something along these lines:

```nix
{ 
  apps.test-ghc8107 = ...; # reference packages.test-ghc8107
  apps.test-ghc902 = ...; # reference packages.test-ghc902
  packages.test-ghc8107 = ...;
  packages.test-ghc902 = ...;
  devShells.ghc8107 = flake-ghc8107.devShells.default;
  devShells.ghc902 = flake-ghc902.devShells.default;
}
```

For the default values for "devShells" and "packages", I am thinking about pointing to the latest GHC version:

```nix
rec {
  ...
  packages.default = flake-ghc902.packages."hapistrano:exe:hap";
  devShells.default = devShells.ghc902;
}
```

Please let me know your thoughts.

Reply:
Looks good @sestrella! Just pushed the changes and the build looks nice! DX seems better now with your suggestion, and even if it could get messy with more versions of ghc, I assume we are not going to support more than 2-3 at the same time.